### PR TITLE
Fixed a few bugs related to closing of tabs

### LINKED ISSUE #1053: [Linux] Close Buttton on Other Tabs Closes the Displayed Tab Instead
On Linux, the close button on tabs seems to only close the currently displayed tab, irrelevant of which tab the close button is located on. The problem is displayed in this video:

https://github.com/kiwix/kiwix-desktop/assets/39098000/a7b748c6-9bc4-446c-8ee5-200476ee7e81

I supposed this was the code that was handling tab closures, and it seems to be the case. This issue doesn't seem to be present on Windows, which means it might be an OS-dependent behavior. The middle button close functionality works fine, and it uses indexes of clicked tabs to close tabs. I have a method to retrieve a tab index (not merged) based on the widget which might be useful #1048? 
Screenshot from 2024-03-07 12-35-58

### LINKED ISSUE #1055: On closing the nth tab, (n+1)th tab becomes the active tab
As shown in the video, i have three tabs open.
When the 1st tab is active tab, and I close the 2nd tab, 3rd tab becomes the active tab.
Expected result: The currently active tab should stay active no matter which tab is closed.
Tab demo

### PR DESCRIPTION
**Pull request by veloman-yunkan**
Fixes #1053
Fixes #1055

Includes some clean-up of related code.

### CODE REVIEW

File: src/tabbar.cpp [hunk 1]
```diff
@@ -151,20 +122,31 @@ void TabBar::moveToPreviousTab()
...
-    } else {
-        index = realTabCount(); // for New Tab Button
-    }
-    mp_stackedWidget->insertWidget(index, tab);
-    index = insertTab(index, "");
+    Q_ASSERT(index > 0 && index < realTabCount());
+
     QToolButton *tb = new QToolButton(this);
-    tb->setDefaultAction(KiwixApp::instance()->getAction(KiwixApp::CloseTabAction));
+    QAction *a = new QAction(QIcon(":/icons/close.svg"), gt("close-tab"), tb);
```
Comment (ShaopengLin) [inline src/tabbar.cpp:130]: The button originally had a tooltip about the keyboard shortcut. Should we include it here as well?

File: src/kiwixapp.cpp [hunk 2]
```diff
@@ -370,8 +382,7 @@ void KiwixApp::createAction()
 
     CREATE_ACTION_ICON_SHORTCUT(NewTabAction,"new-tab-icon", gt("new-tab"), QKeySequence::AddTab);
 
-    CREATE_ACTION_ICON_SHORTCUTS(CloseTabAction, "close", gt("close-tab"), QList<QKeySequence>({QKeySequence(Qt::CTRL | Qt::Key_F4), QKeySequence(Qt::CTRL | Qt::Key_W)}));
-    mpa_actions[CloseTabAction]->setIconVisibleInMenu(false);
+    CREATE_ACTION_SHORTCUTS(CloseCurrentTabAction, gt("close-tab"), QList<QKeySequence>({QKeySequence(Qt::CTRL | Qt::Key_F4), QKeySequence(Qt::CTRL | Qt::Key_W)}));
```
Comment (ShaopengLin) [inline src/kiwixapp.cpp:385]: Should we be more explicit that it only closes the current Tab?

File: src/tabbar.cpp [hunk 3]
```diff
@@ -265,48 +247,43 @@ void TabBar::triggerWebPageAction(QWebEnginePage::WebAction action, ZimView *wid
...
+    // cannot be closed
+    if (index <= 0 || index == this->realTabCount())
         return;
 
-    setSelectionBehaviorOnRemove(index);
-
     QWidget *view = mp_stackedWidget->widget(index);
 
     // library tab cannot be closed
     if (qobject_cast<ContentManagerView*>(view)) {
```
Comment (ShaopengLin) [inline src/tabbar.cpp:272]: This is no longer relevant if you added the check above.

**Review by ShaopengLin — COMMENTED** [review]
LGTM, only a few minor questions.

File: src/tabbar.cpp (hunk 1 above)
Comment (veloman-yunkan) [reply src/tabbar.cpp:138]: Done

File: src/kiwixapp.cpp (hunk 2 above)
Comment (veloman-yunkan) [reply src/kiwixapp.cpp:385]: I don't think that any other interpretation of this text in the UI could make sense

**Review by mgautierfr — COMMENTED** [review]
LGTM.

Please rebase-fixup before final approval.
You question is interesting. I let @kelson42 jump into the discussion but it is not a blocker for this PR.

**Comment by veloman-yunkan** [comment]
> Please rebase-fixup before final approval.

Done

**Comment by kelson42** [comment]
I will answer in dedicated ticket. We are good to merge IMO.

**Pull request merged**